chpl/chpl-service/src/main/java/gov/healthit/chpl/caching/AsynchronousCacheInitialization.java

```diff
@@ -0,0 +1,124 @@
+package gov.healthit.chpl.caching;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import gov.healthit.chpl.dao.CertificationStatusDAO;
+import gov.healthit.chpl.dao.CertifiedProductSearchResultDAO;
+import gov.healthit.chpl.dao.EntityRetrievalException;
+import gov.healthit.chpl.dao.PendingCertifiedProductDAO;
+import gov.healthit.chpl.domain.SearchRequest;
+import gov.healthit.chpl.domain.SurveillanceSearchOptions;
+import gov.healthit.chpl.dto.CertificationStatusDTO;
+import gov.healthit.chpl.manager.CertificationIdManager;
+import gov.healthit.chpl.manager.SearchMenuManager;
+
+@Component
+public class AsynchronousCacheInitialization {
+	private static final Logger logger = LogManager.getLogger(AsynchronousCacheInitialization.class);
+	
+	@Autowired private CertificationIdManager certificationIdManager;
+	@Autowired private SearchMenuManager searchMenuManager;
+	@Autowired private CertifiedProductSearchResultDAO certifiedProductSearchResultDAO;
+	@Autowired private PendingCertifiedProductDAO pcpDao;
+	@Autowired private CertificationStatusDAO statusDao;
+	
+	@Async
+	@Transactional
+	public void initializeSearchOptions() throws EntityRetrievalException{
+		logger.info("Starting cache initialization for SearchViewController.getPopulateSearchData()");
+		searchMenuManager.getCertBodyNames();
+		searchMenuManager.getEditionNames(false);
+		searchMenuManager.getEditionNames(true);
+		searchMenuManager.getCertificationStatuses();
+		searchMenuManager.getPracticeTypeNames();
+		searchMenuManager.getClassificationNames();
+		searchMenuManager.getProductNames();
+		searchMenuManager.getDeveloperNames();
+		searchMenuManager.getCQMCriterionNumbers(false);
+		searchMenuManager.getCQMCriterionNumbers(true);
+		searchMenuManager.getCertificationCriterionNumbers(false);
+		searchMenuManager.getCertificationCriterionNumbers(true);
+		logger.info("Finished cache initialization for SearchViewController.getPopulateSearchData()");
+	}
+	
+	@Async
+	@Transactional
+	public void initializePending() throws EntityRetrievalException{
+		logger.info("Starting cache initialization for /pending");
+		CertificationStatusDTO statusDto = statusDao.getByStatusName("Pending");
+		pcpDao.findByStatus(statusDto.getId());
+		logger.info("Finished cache initialization for /pending");
+	}
+	
+	@Async
+	@Transactional
+	public void initializeSearch() throws EntityRetrievalException{
+		logger.info("Starting cache initialization for /search DAO method");
```

Comment:
I'd definitely like to cache the default search, but maybe we can find a way to put the "default search" as an external properties file? To make updating it in the future easier? If there's no way of doing that I'd still lean towards caching, even with hard-coded values, and maybe add some comments to the /search call somewhere to remind us to update the cache if those "defaults" change